File: front-end/src/app/service-models/app/v1/client/client-employee-document-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { EmployeeDocumentTypeSM } from '../../enums/employee-document-type-s-m.enum';

export class ClientEmployeeDocumentSM extends CoinManagementServiceModelBase<number> {
    clientUserId!: number;
    clientCompanyDetailId!: number;
    name!: string;
    employeeDocumentPath!: string;
    employeeDocumentType!: EmployeeDocumentTypeSM;
    documentDescription!: string;
    extension!: string;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientEmployeeDocumentSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientEmployeeDocumentSM : SiffrumPayrollServiceModelBase<int>
    {
        public int ClientUserId { get; set; }
        public int ClientCompanyDetailId { get; set; }
        public string Name { get; set; }
        public string EmployeeDocumentPath { get; set; }
        public EmployeeDocumentTypeSM EmployeeDocumentType { get; set; }
        public string DocumentDescription { get; set; }
        public string Extension { get; set; }

    }
}

